mesh/src/main/java/no/nordicsemi/android/mesh/transport/GenericBatteryStatus.java

```diff
@@ -40,9 +40,9 @@ void parseStatusParameters() {
         final ByteBuffer buffer = ByteBuffer.wrap(mParameters).order(ByteOrder.LITTLE_ENDIAN);
         mBatteryLevel = buffer.get();
         MeshLogger.verbose(TAG, "Battery level: " + mBatteryLevel);
-        if (buffer.limit() > GENERIC_BATTERY_STATUS_MANDATORY_LENGTH) {
-            mTimeToDischarge = buffer.get() | (buffer.get() << 8) | (buffer.get() << 16);
-            mTimeToCharge = buffer.get() | (buffer.get() << 8) | (buffer.get() << 16);
+        if (buffer.limit() >= GENERIC_BATTERY_STATUS_MANDATORY_LENGTH) {
+            mTimeToDischarge = (buffer.get() & 0xFF) | ((buffer.get() & 0xFF) << 8) | ((buffer.get() & 0xFF) << 16);
+            mTimeToCharge = (buffer.get() & 0xFF) | ((buffer.get() & 0xFF)<< 8) | ((buffer.get() & 0xFF) << 16);
```

Comment:
Once we get rid of the bytebuffer, we could directly work with the `mParameters` 
`(mParameters[1] & 0xFF) | ((mParameters[2] & 0xFF) << 8) | (mParameters[3] & 0xFF) << 16)` and so forth.

Reply:
👍